### PR TITLE
Extended modulemd.h annotations

### PR DESCRIPTION
**Pull request by ananyaarun**
Created Python tutorial to list the RPM API for a provided module NSVC by extending modulemd.h annotations.

### CODE REVIEW

File: modulemd/v2/include/modulemd-2.0/modulemd.h [hunk 1]
```diff
@@ -172,9 +172,64 @@ G_BEGIN_DECLS
...
+ * This is done as follows
+ *
+ * |[<!-- language="Python" -->
+ * module = fedora_index.get_module ('modulename')
+ * defaults = module.get_defaults()
+ * print ('Default stream for modulename is %s' % (
+ *        defaults.get_default_stream())
+ * ]|
+ *
+ * Fourth step is to Query the Modulemd.Module for the Modulemd.ModuleStream
```
Comment (sgallagher) [inline modulemd/v2/include/modulemd-2.0/modulemd.h]:
```suggestion
 * The fourth step is to query the Modulemd.Module for the Modulemd.ModuleStream
```

File: modulemd/v2/include/modulemd-2.0/modulemd.h [hunk 2]
```diff
@@ -172,9 +172,64 @@ G_BEGIN_DECLS
...
+ * associated with the provided NSVC (name-stream-version-context identifier)
+ * This is done as follows
+ *
+ * |[<!-- language="Python" -->
+ * stream = module.get_stream_by_NSVC('modulestream', 1, 'deadbeef')
+ * ]|
+ *
+ * Last step is to Read the RPM API from the Modulemd.ModuleStream
```
Comment (sgallagher) [inline modulemd/v2/include/modulemd-2.0/modulemd.h]:
```suggestion
 * The last step is to read the RPM API from the Modulemd.ModuleStream
```

File: modulemd/v2/include/modulemd-2.0/modulemd.h [hunk 3]
```diff
@@ -172,9 +172,64 @@ G_BEGIN_DECLS
...
+ * This is done as follows
+ *
+ * |[<!-- language="Python" -->
+ * api_list = stream.get_rpm_api()
```
Comment (sgallagher) [inline modulemd/v2/include/modulemd-2.0/modulemd.h]: Describe that the `api_list` is a python list of strings containing package names.

File: modulemd/v2/include/modulemd-2.0/modulemd.h [hunk 4]
```diff
@@ -172,9 +172,64 @@ G_BEGIN_DECLS
...
+ * ]|
+ *
+ * In the example above we Imported the libmodule 2.0 API
```
Comment (sgallagher) [inline modulemd/v2/include/modulemd-2.0/modulemd.h]: Drop this conclusion, it's redundant.

**Comment by ananyaarun** [comment]
@sgallagher I made the required changes 
But i get this error when i run ninja to build the sources which im still trying to figure out.
"ninja: error: loading 'build.ninja': No such file or directory" 
The meson command works fine and i can see the api2 subdirectory created.

**Comment by sgallagher** [comment]
> @sgallagher I made the required changes
> But i get this error when i run ninja to build the sources which im still trying to figure out.
> "ninja: error: loading 'build.ninja': No such file or directory"
> The meson command works fine and i can see the api2 subdirectory created.

You need to run the `ninja` command from within the api2 subdirectory (or else as `ninja -C api2`)

**Comment by ananyaarun** [comment]
@sgallagher i am able to run ninja now but am facing these issues !
1) After running ninja i got this error,

`ERROR: can't resolve libraries to shared libraries: yaml
ninja: build stopped: subcommand failed.`

So i tried upgrading my gobject-introspection version to gobject-introspection >= 1.54

2) While doing so i ran into this other problem,
`configure: error: Package requirements (glib-2.0 >= 2.58.0) were not met:

Requested 'glib-2.0 >= 2.58.0' but version of GLib is 2.48.2

Consider adjusting the PKG_CONFIG_PATH environment variable if you
installed software in a non-standard prefix.

Alternatively, you may set the environment variables GLIB_CFLAGS
and GLIB_LIBS to avoid the need to call pkg-config.
See the pkg-config man page for more details.`

Im not sure why this is happening.

**Comment by sgallagher** [comment]
> @sgallagher i am able to run ninja now but am facing these issues !
> 
> 1. After running ninja i got this error,
> 
> `ERROR: can't resolve libraries to shared libraries: yaml ninja: build stopped: subcommand failed.`
> 
> So i tried upgrading my gobject-introspection version to gobject-introspection >= 1.54
> 
> 1. While doing so i ran into this other problem,
>    `configure: error: Package requirements (glib-2.0 >= 2.58.0) were not met: Requested 'glib-2.0 >= 2.58.0' but version of GLib is 2.48.2 Consider adjusting the PKG_CONFIG_PATH environment variable if you installed software in a non-standard prefix. Alternatively, you may set the environment variables GLIB_CFLAGS and GLIB_LIBS to avoid the need to call pkg-config. See the pkg-config man page for more details.`
> 
> Im not sure why this is happening.

What OS are you running? You need to be on Fedora 28 or later.

**Comment by sgallagher** [comment]
@ananyaarun Are you still planning to work on this?

**Comment by OrionStar25** [comment]
Since there has been no update on this for > 20 days, I'll work on this. I'll pick up from where she left.

**Pull request closed without merging**